test/e2e/nsx_subnet_test.go

```diff
@@ -602,3 +699,574 @@ func SubnetValidate(t *testing.T) {
+func verifyNoSharedSubnet(t *testing.T, namespace string) {
+	verifySharedSubnets(t, namespace, 0)
+}
+
+// verifySharedSubnet verifies that a shared subnet with the expected name prefix exists in the namespace
+func verifySharedSubnet(t *testing.T, namespace string, namePrefix string) *v1alpha1.Subnet {
+	subnets := verifySharedSubnets(t, namespace, 1, namePrefix)
+	return subnets[0]
+}
+
+// verifyMultipleSharedSubnets verifies that multiple shared subnets with the expected name prefixes exist in the namespace
+func verifyMultipleSharedSubnets(t *testing.T, namespace string, expectedCount int, namePrefixes ...string) []*v1alpha1.Subnet {
+	return verifySharedSubnets(t, namespace, expectedCount, namePrefixes...)
+}
+
+// PrecreatedSharedSubnetBasic tests sharing a subnet with multiple namespaces
+func PrecreatedSharedSubnetBasic(t *testing.T) {
+	// The subnet is created directly in the target namespace using the NSX client REST API
+	// because the nsx-operator will append an underscore to the subnet name.
+	// When retrieving this name from nsx to create the CR again,
+	// the presence of an underscore causes the CR creation to fail
+	subnetName := "shared-subnet"
+	subnetPath, _ := createSharedSubnet(t, subnetName)
+	// Update ns1 VPC networkconfig CR with shared subnet
+	updateVPCNetworkConfigWithSubnet(t, ns1, subnetPath)
+	// Update ns2 VPC networkconfig CR with shared subnet
+	updateVPCNetworkConfigWithSubnet(t, ns2, subnetPath)
+	// Verify the shared subnets exist in ns1 and ns2 with correct properties
+	sharedSubnet1 := verifySharedSubnet(t, ns1, subnetName)
+	sharedSubnet2 := verifySharedSubnet(t, ns2, subnetName)
+	log.Info("Shared subnet verification complete", "ns1_subnet", sharedSubnet1.Name, "ns2_subnet", sharedSubnet2.Name)
+}
+
+// PrecreatedSharedSubnetRemovePath tests removing a shared subnet path from one namespace
+func PrecreatedSharedSubnetRemovePath(t *testing.T) {
+	// Remove the shared subnet path from ns1 vpcNetworkConfig
+	// ns1 should not have the shared subnet anymore, but ns2 should still have it
+	clearVPCNetworkConfigSubnets(t, ns1)
+	// Verify the shared subnet is removed from ns1
+	verifyNoSharedSubnet(t, ns1)
+	// Verify ns2 still has the shared subnet with correct properties
+	sharedSubnet2 := verifySharedSubnet(t, ns2, "shared-subnet")
+	log.Info("Shared subnet verification after removal complete", "ns2_subnet", sharedSubnet2.Name)
+}
+
+// updateSubnetConnectivityState updates a subnet's connectivity state in NSX
+func updateSubnetConnectivityState(t *testing.T, namespace, subnetName, connectivityState string) {
+	// Find the subnet in NSX by name and namespace
+	subnets, subnetPath := testData.fetchSubnetByNameAndNamespace(t, subnetName, namespace)
+	require.NotEmpty(t, subnets, "No subnet found with name %s in namespace %s", subnetName, namespace)
+	require.NotEmpty(t, subnetPath, "Subnet path is empty for subnet %s in namespace %s", subnetName, namespace)
+
+	// Extract the patch URL from the path
```

Comment:
it seems more easy to use SubnetClient.Update based on subnet fetched by fetchSubnetByNameAndNamespace.
you fetched the subnet twice. one is by query, the other is by Get.

Reply:
fixed